### PR TITLE
rewrite trio_matrix in Python.

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

**Comment by cseed** [comment]
This exposes a bug in the pruner in test_tdt that I'm still investigating.

File: hail/python/hail/methods/family_methods.py [hunk 1]
```diff
@@ -59,7 +60,55 @@ def trio_matrix(dataset, pedigree, complete_trios=False) -> MatrixTable:
     -------
     :class:`.MatrixTable`
     """
-    return MatrixTable._from_java(dataset._jmt.trioMatrix(pedigree._jrep, complete_trios))
+    mt = dataset
+    if not (len(mt.col_key) == 1 and mt.col_key[0].dtype == hl.tstr):
```
Comment (tpoterba) [inline hail/python/hail/methods/family_methods.py]: use `check_col_key_str`

File: hail/python/hail/methods/family_methods.py [hunk 2]
```diff
@@ -59,7 +60,55 @@ def trio_matrix(dataset, pedigree, complete_trios=False) -> MatrixTable:
...
+        raise ValueError(f'trio_matrix requires column key of type [str], found: {",".join(list(mt.colkey))}')
+    
+    k = mt.col_key.dtype.fields[0]
+    samples = mt[k].collect()
```
Comment (tpoterba) [inline hail/python/hail/methods/family_methods.py]: can just be `mt.col_key[0].collect()`

**Comment by tpoterba** [comment]
tdt is failing here, right? I can go chase after that.

**Comment by cseed** [comment]
Yes, but I think I know what's wrong.  Taking a swing at it now.  I'll let you know if I need help.

File: hail/python/hail/methods/family_methods.py (hunk 2 above)
Comment (cseed) [reply hail/python/hail/methods/family_methods.py]: I need `k` below.  I don't think changing the collect line is clearer.

Comment (tpoterba) [reply hail/python/hail/methods/family_methods.py]: ah, right.

**Pull request merged**